src/main/java/com/ebivariation/contigalias/service/AssemblyService.java

```diff
@@ -18,27 +18,90 @@
 
 import com.ebivariation.contigalias.dao.AssemblyDao;
 import com.ebivariation.contigalias.entities.AssemblyEntity;
+import com.ebivariation.contigalias.entities.ChromosomeEntity;
+import com.ebivariation.contigalias.repo.AssemblyRepository;
+import org.hibernate.LazyInitializationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
 import java.io.IOException;
+import java.util.List;
 import java.util.Optional;
 
 @Service
 public class AssemblyService {
 
-    private final AssemblyDao assemblyDao;
+    private final AssemblyRepository repository;
+
+    private final AssemblyDao dao;
+
+    private final Logger logger = LoggerFactory.getLogger(AssemblyService.class);
 
     @Autowired
-    public AssemblyService(@Qualifier("ftpDao") AssemblyDao assemblyDao) {
-        this.assemblyDao = assemblyDao;
+    public AssemblyService(AssemblyRepository repository, AssemblyDao dao) {
+        this.repository = repository;
+        this.dao = dao;
+    }
+
+    public Optional<AssemblyEntity> getAssemblyOrFetchByAccession(String accession) {
+        Optional<AssemblyEntity> assembly = getAssemblyByAccession(accession);
+        if (assembly.isPresent()) {
+            return assembly;
+        }
+        try {
+            Optional<AssemblyEntity> fetchAssembly = dao.getAssemblyByAccession(accession);
+            if (fetchAssembly.isPresent()) {
+                AssemblyEntity asm = fetchAssembly.get();
+                insertAssembly(asm);
+                List<ChromosomeEntity> chromosomes = asm.getChromosomes();
+                if (chromosomes != null) {
+                    chromosomes.forEach(chr -> chr.setAssembly(null));
+                }
+                return fetchAssembly;
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return Optional.empty();
     }
 
-    public Optional<AssemblyEntity> getAssemblyByAccession(String accession) throws IOException {
-        Optional<AssemblyEntity> assembly = assemblyDao.getAssemblyByAccession(accession);
-        assembly.ifPresent(asm -> asm.getChromosomes().forEach(chr -> chr.setAssembly(null)));
+    public Optional<AssemblyEntity> getAssemblyByAccession(String accession) {
+        Optional<AssemblyEntity> assembly = repository.dFindAssemblyEntityByAccession(accession);
+        assembly.ifPresent(asm -> {
+            try {
+                List<ChromosomeEntity> chromosomes = asm.getChromosomes();
+                if (chromosomes != null && chromosomes.size() > 0) {
+                    chromosomes.forEach(chr -> chr.setAssembly(null));
+                }
+            } catch (LazyInitializationException e) {
```

Comment:
I haven't used hibernate much, but it seems that if you get this exception, you are making a lazy query and expecting the relations (joins) to be made. If this happens, it should probably fail as loud as possible to tell us that we have to change something.
If you really want to log the message here, you can log and rethrow.

Reply:
I did some research and got rid of `LazyInitializationException` by adding the `@Transactional` annotatio. I'm not sure if that disables lazy load which could be a problem when each entity has hundreds of scaffolds associated with it.

Reply:
for now it's ok to use EAGER to see the data, but when we have some more endpoints, all the queries will be lazy.